src/runtime/virtcontainers/container.go

```diff
@@ -1075,38 +1080,68 @@ func (c *Container) annotateContainerWithVFIOMetadata(devices interface{}) {
 			siblings[i].Index = i
 		}
 
-		// Now that we have the index lets connect the /dev/vfio/<num>
-		// to the correct index
-		if devices, ok := devices.([]ContainerDevice); ok {
-			for _, dev := range devices {
-				c.siblingAnnotation(dev.ContainerPath, siblings)
-			}
-		}
+		// Remove all outer runtime CDI annotations (cdi.k8s.io/*) from the container spec
+		// before creating guest-specific (inner runtime) CDI annotations. The host-side
+		// annotations (e.g., cdi.k8s.io/gpu: nvidia.com/pgpu=0) served their purpose for
+		// device allocation and cold-plugging, but reference device kinds that don't exist
+		// in the guest where they would be unresolvable.
+		c.removeOuterCDIAnnotations()
 
-		if devices, ok := devices.([]config.DeviceInfo); ok {
-			for _, dev := range devices {
-				c.siblingAnnotation(dev.ContainerPath, siblings)
-			}
+		// Create guest-compatible CDI annotations by mapping each /dev/vfio/<num> device
+		// to its indexed sibling (e.g., cdi.k8s.io/vfio50: nvidia.com/gpu=0).
+		c.addInnerCDIAnnotations(devices, siblings)
+	}
+}
 
+func (c *Container) addInnerCDIAnnotations(devices interface{}, siblings []DeviceRelation) {
```

Comment:
I think we should avoid use the *Inner* and *Outer* in the code, this is not a commen sense for users, which will make users confused.